otelcol/otelcoltest/config.go

```diff
@@ -30,15 +41,33 @@ func LoadConfig(fileName string, factories otelcol.Factories) (*otelcol.Config,
 			ConverterFactories: []confmap.ConverterFactory{expandconverter.NewFactory()},
 		},
 	})
+}
+
+// LoadConfigWithSettingsAndValidate loads a config from the provider settings, and validates the configuration.
+func LoadConfigWithSettingsAndValidate(factories otelcol.Factories, set otelcol.ConfigProviderSettings) (*otelcol.Config, error) {
```

Comment:
Nit: usually the "WithSettings" would be a suffix to the method. This makes swapping the function in code a bit easier (if you're not using just straight up search and replace)

Reply:
Updated